src/librustdoc/test.rs

```diff
@@ -341,23 +345,27 @@ pub fn make_test(s: &str,
         // commonly used to make tests fail in case they trigger warnings, so having this there in
         // that case may cause some tests to pass when they shouldn't have.
         prog.push_str("#![allow(unused)]\n");
+        line_offset += 1;
     }
 
     // Next, any attributes that came from the crate root via #![doc(test(attr(...)))].
     for attr in &opts.attrs {
         prog.push_str(&format!("#![{}]\n", attr));
+        line_offset += 1;
     }
 
     // Now push any outer attributes from the example, assuming they
     // are intended to be crate attributes.
     prog.push_str(&crate_attrs);
+    line_offset += crate_attrs.lines().count();
```

Comment:
Is `line_offset` supposed to be the number of lines inserted by rustdoc? IIRC, `crate_attrs` is actually the `#![]` attributes from the example itself.

Reply:
Yeah, you're right. Given that the attrs can only be up top we should just not count this.